### PR TITLE
Feature/core 8451 geo list item emiting events when is disabled

### PR DESCRIPTION
**Pull request by asanzdj**
**Jira task:**

https://geoblink.atlassian.net/browse/CORE-8451

**Related info to solve the task:**

- Vue $listeners: https://vuejs.org/v2/api/#vm-listeners
- Combine $listeners with control events: https://github.com/vuejs/vue/issues/7042
- Transparent components ($listeners + custom events): https://zendev.com/2018/05/31/transparent-wrapper-components-in-vue.html
- Functional components: https://vue-loader.vuejs.org/guide/functional.html, https://v3.vuejs.org/guide/migration/functional-components.html

### CODE REVIEW

File: src/elements/GeoList/GeoListItem.vue [hunk 1]
```diff
@@ -1,22 +1,22 @@
-<template functional>
+<template>
```
Comment (Navas23) [inline src/elements/GeoList/GeoListItem.vue:1]: Why did you remove the functional?

File: src/elements/GeoList/GeoListItem.vue [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
   <component
-    :is="props.wrapperTag"
-    :ref="data.ref"
-    v-bind="data.attrs"
-    :class="[data.class, data.staticClass, {
+    :is="wrapperTag"
+    :ref="$attrs.ref"
```
Comment (Navas23) [inline src/elements/GeoList/GeoListItem.vue:4]: This is because it's not functional, right?

File: src/elements/GeoSegmentedControl/GeoInfoSegmentedControlItem.vue [hunk 3]
```diff
@@ -28,9 +29,36 @@ export default {
...
+      default: false
+    },
+
+    /**
+     * When is true the component will display an active style.
+     */
+    active: {
+      type: Boolean,
+      default: false
+    }
```
Comment (fofi) [inline src/elements/GeoSegmentedControl/GeoInfoSegmentedControlItem.vue:61]: Why are we adding these props if they are already in the mixin?

File: src/elements/GeoSegmentedControl/GeoInfoSegmentedControlItem.vue [hunk 4]
```diff
@@ -1,16 +1,16 @@
-<template functional>
+<template>
```
Comment (fofi) [inline src/elements/GeoSegmentedControl/GeoInfoSegmentedControlItem.vue:1]: We had the code like this because this was a functional component, probably now that is not anymore, we can modify it like a normal component

**Review by fofi — COMMENTED** [review]
Should we add this to other components?

File: src/elements/GeoList/GeoListItem.vue (hunk 1 above)
Comment (asanzdj) [reply src/elements/GeoList/GeoListItem.vue:1]: If it's functional I can't take control about its events.

File: src/elements/GeoList/GeoListItem.vue (hunk 2 above)
Comment (asanzdj) [reply src/elements/GeoList/GeoListItem.vue:4]: Yes

File: src/elements/GeoSegmentedControl/GeoInfoSegmentedControlItem.vue (hunk 4 above)
Comment (asanzdj) [reply src/elements/GeoSegmentedControl/GeoInfoSegmentedControlItem.vue:1]: Ah, okey, I'll try it

File: src/elements/GeoSegmentedControl/GeoInfoSegmentedControlItem.vue (hunk 3 above)
Comment (asanzdj) [reply src/elements/GeoSegmentedControl/GeoInfoSegmentedControlItem.vue]: Because I didn't notice about that but I've already removed then

File: src/elements/GeoList/GeoListItem.vue [hunk 5]
```diff
@@ -1,22 +1,21 @@
-<template functional>
+<template>
   <component
-    :is="props.wrapperTag"
-    :ref="data.ref"
```
Comment (Navas23) [inline src/elements/GeoList/GeoListItem.vue:4]: We don't need this ref now?

File: test/unit/specs/elements/GeoList/GeoListItem.spec.js [hunk 6]
```diff
@@ -1,19 +1,18 @@
...
+import { shallowMount } from '@vue/test-utils'
 import GeoListItem from '@/elements/GeoList/GeoListItem.vue'
 import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
 import { library } from '@fortawesome/fontawesome-svg-core'
 import { fas } from '@fortawesome/free-solid-svg-icons'
 
 library.add(fas)
 
-describe('GeoListItem', () => {
+describe.only('GeoListItem', () => {
```
Comment (Navas23) [inline test/unit/specs/elements/GeoList/GeoListItem.spec.js:9]: Why do we need the only here?

File: test/unit/specs/elements/GeoList/GeoListItem.spec.js [hunk 7]
```diff
@@ -22,42 +21,8 @@ describe('GeoListItem', () => {
...
-  it('Should render with the provided static class', () => {
-    const wrapper = mount(GeoListItem, {
-      context: {
-        staticClass: 'my-static-class'
-      }
-    })
-    expect(wrapper.find('div.geo-list-item.my-static-class').exists()).toBe(true)
-  })
-
-  it('Should render with the provided dynamic class', () => {
```
Comment (Navas23) [inline test/unit/specs/elements/GeoList/GeoListItem.spec.js:34]: Why did you remove so many tests?

File: src/elements/GeoList/GeoListItem.vue (hunk 5 above)
Comment (asanzdj) [reply src/elements/GeoList/GeoListItem.vue:4]: I did some tries to know if passing a ref is necessary or not. And I've noticed that is not needed.
For example, here, in the live example I have that ref without receiving it as a prop in the component.

Screenshot 2020-11-03 at 12 38 42

File: test/unit/specs/elements/GeoList/GeoListItem.spec.js (hunk 6 above)
Comment (asanzdj) [reply test/unit/specs/elements/GeoList/GeoListItem.spec.js:9]: I'm fixing test and  I've forgotten about this

File: test/unit/specs/elements/GeoList/GeoListItem.spec.js (hunk 7 above)
Comment (asanzdj) [reply test/unit/specs/elements/GeoList/GeoListItem.spec.js:34]: This tests was associated to a weird behaviour, dynamic and static classes. I removed that behaviour so I removed related tests too. I noticed that you can add extra classes to the component instance to style it and it isn't needed to receive them as a prop by the component.

**Review by Navas23 — APPROVED** [review]

File: test/unit/specs/elements/GeoList/GeoListItem.spec.js [hunk 8]
```diff
@@ -104,22 +67,20 @@ describe('GeoListItem', () => {
 
   it('Should emit click event', function () {
```
Comment (fofi) [inline test/unit/specs/elements/GeoList/GeoListItem.spec.js:68]: We should create a test testing the disabled not emitting, right?

File: test/unit/specs/elements/GeoSegmentedControl/GeoSegmentedControlItem.spec.js [hunk 9]
```diff
@@ -30,37 +30,37 @@ describe('GeoSegmentedControlItem', () => {
...
-        expect(wrapper.emitted().click).toBeTruthy()
-        done()
-      } catch (error) {
-        done(error)
-      }
-    })
+    expect(clickListener).toHaveBeenCalled()
   })
 
   it('Should not emit an event when it\'s disabled', function () {
```
Comment (fofi) [inline test/unit/specs/elements/GeoSegmentedControl/GeoSegmentedControlItem.spec.js:47]: Why Was this test working perfectly?

File: test/unit/specs/elements/GeoList/GeoListItem.spec.js (hunk 8 above)
Comment (asanzdj) [reply test/unit/specs/elements/GeoList/GeoListItem.spec.js:68]: Added

**Review by fofi — APPROVED** [review]
